File: Leetcode/dynamic_programming/338_Counting Bits/brute_force/Solution.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        def count(num):
            res = 0
            while num:
                res += num%2
                num = num//2
            return res
        res = [count(i) for i in range(num+1)]
        return res

File: Leetcode/dynamic_programming/338_Counting Bits/brute_force/Solution.java
class Solution {
    public int[] countBits(int num) {
        int[] res = new int[num+1];
        for(int i = 0; i <= num; i++){
            res[i] = bits(i);
        }
        return res;
    }
        private int bits(int n) {
            int count = 0;
            while(n!=0){
                n = (n&(n-1));
                count++;
            }
            return count;
        }
}
